Examples/DpdkBridge/main.cpp

```diff
@@ -0,0 +1,336 @@
+	TablePrinter printer(columnNames, columnWidths);
+
+	// print final stats for every worker thread plus sum of all threads and free worker threads memory
+	PacketStats aggregatedStats;
+	for (std::vector<DpdkWorkerThread*>::iterator iter = args->workerThreadsVector->begin(); iter != args->workerThreadsVector->end(); iter++)
+	{
+		AppWorkerThread* thread = (AppWorkerThread*)(*iter);
+		PacketStats threadStats = thread->getStats();
+		aggregatedStats.collectStats(threadStats);
+		printer.printRow(threadStats.getStatValuesAsString("|"), '|');
+		delete thread;
+	}
+
+	printer.printSeparator();
+	printer.printRow(aggregatedStats.getStatValuesAsString("|"), '|');
+
+	args->shouldStop = true;
+}
+
+
+/**
+ * main method of the application. Responsible for parsing user args, preparing worker thread configuration, creating the worker threads and activate them.
+ * At program termination worker threads are stopped, statistics are collected from them and printed to console
+ */
+int main(int argc, char* argv[])
+{
+	AppName::init(argc, argv);
+
+	std::vector<int> dpdkPortVec;
+
+	CoreMask coreMaskToUse = getCoreMaskForAllMachineCores();
+
+	int sendPacketsToPort = -1;
+
+	int optionIndex = 0;
+	char opt = 0;
+
+	uint32_t mBufPoolSize = DEFAULT_MBUF_POOL_SIZE;
+
+	while((opt = getopt_long (argc, argv, "d:c:m:hvl", FilterTrafficOptions, &optionIndex)) != -1)
+	{
+		switch (opt)
+		{
+			case 0:
+			{
+				break;
+			}
+			case 'd':
+			{
+				string portListAsString = string(optarg);
+				stringstream stream(portListAsString);
+				string portAsString;
+				int port;
+				// break comma-separated string into string list
+				while(getline(stream, portAsString, ','))
+				{
+					char c;
+					std::stringstream stream2(portAsString);
+					stream2 >> port;
+					if (stream2.fail() || stream2.get(c))
+					{
+						// not an integer
+						EXIT_WITH_ERROR_AND_PRINT_USAGE("DPDK ports list is invalid");
+					}
+					dpdkPortVec.push_back(port);
+				}
+				// verify list contains two ports
+				if(dpdkPortVec.size()!=2)
+				{
+					EXIT_WITH_ERROR_AND_PRINT_USAGE("DPDK list must contain two values");
+				}
+				break;
+			}
+			case 'c':
+			{
+				coreMaskToUse = atoi(optarg);
+				break;
+			}
+			case 'm':
+			{
+				mBufPoolSize = atoi(optarg);
+				break;
+			}
+			case 'h':
+			{
+				printUsage();
+				exit(0);
+			}
+			case 'v':
+			{
+				printAppVersion();
+				break;
+			}
+			case 'l':
+			{
+				listDpdkPorts();
+				exit(0);
+			}
+			default:
+			{
+				printUsage();
+				exit(0);
+			}
+		}
+	}
+
+	// verify list is not empty
+	if (dpdkPortVec.empty())
+	{
+		EXIT_WITH_ERROR_AND_PRINT_USAGE("DPDK ports list is empty. Please use the -d switch");
+	}
+
+	// extract core vector from core mask
+	vector<SystemCore> coresToUse;
+	createCoreVectorFromCoreMask(coreMaskToUse, coresToUse);
+
+	// need minimum of 2 cores to start - 1 management core + 1 (or more) worker thread(s)
+	if (coresToUse.size() < 2)
```

Comment:
Need to check a minimum of 3 cores - 1 for each of the two workers and 1 for management

Reply:
Agreed, I fixed it